db.js

```diff
@@ -308,15 +343,44 @@ var loans = (function () {
       return loanData;
     },
 
-    updateLoanById: function (loan) {
+    insert: async function (loan) {
       var sql =
-        "UPDATE loans SET requester_id = ?, book_id = ?, owner_contact = ?, " +
-        "requester_contact = ?, request_date = ?, accept_date = ?, loan_start_date = ?, " +
-        "loan_end_date = ?, return_date = ?, status = ? " +
-        "WHERE id = ?";
-      var inserts = [
+        "INSERT INTO loans (requester_id, book_id, requester_contact, owner_contact, status) " +
+        "VALUES (?,?,?,?,?)";
+      var initialInserts = [
         loan.requester_id,
         loan.book_id,
+        loan.requester_contact,
+        " ",
+        "pending",
+      ];
+      sql = mysql.format(sql, initialInserts);
```

Comment:
I think it would make more sense to move the above code (lines 347-357) to after the loan count is finished, and it's verified that the insert is allowed.